Error compiling with --all-features

I tried to compile divviup-api with `--all-features`:

```
$ cargo build --release --all-features
[... snip compilation ...]
error[E0599]: no method named `provider` found for struct `opentelemetry_sdk::trace::TracerProvider` in the current scope
  --> src/telemetry.rs:52:14
   |
46 | /         opentelemetry_otlp::new_pipeline()
47 | |             .tracing()
48 | |             .with_trace_config(opentelemetry_sdk::trace::Config::default().with_resource(resource))
49 | |             .with_exporter(opentelemetry_otlp::new_exporter().tonic())
50 | |             .install_batch(opentelemetry_sdk::runtime::Tokio)
51 | |             .unwrap()
52 | |             .provider()
   | |             -^^^^^^^^ method not found in `TracerProvider`
   | |_____________|
   |

For more information about this error, try `rustc --explain E0599`.
error: could not compile `divviup-api` (lib) due to 1 previous error
```

I'm too much of a Rust noob to understand or provide a patch. :sweat_smile:

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -42,6 +42,8 @@ jobs:
         run: cargo fmt --all -- --check
       - name: check +api-mocks
         run: cargo check --features api-mocks
+      - name: check +otlp-trace
+        run: cargo check --features otlp-trace
       - name: check cli -default +ring
         run: cargo check -p divviup-cli --no-default-features --features ring,common
       - name: check cli +default +admin

--- a/src/telemetry.rs
+++ b/src/telemetry.rs
@@ -48,8 +48,6 @@ pub fn metrics_exporter() -> Result<impl trillium::Handler, MetricsError> {
             .with_trace_config(opentelemetry_sdk::trace::Config::default().with_resource(resource))
             .with_exporter(opentelemetry_otlp::new_exporter().tonic())
             .install_batch(opentelemetry_sdk::runtime::Tokio)
-            .unwrap()
-            .provider()
             .unwrap(),
     );
 

